### PR TITLE
Fix shadow table to detect empty passwords

### PR DESCRIPTION
**Pull request by directionless**
Add branch to if statement to detect empty, but not null, passwords.

I don't see how to easily add test cases here, and can't readily test this

Fixes: #5465

### CODE REVIEW

File: osquery/tables/system/linux/shadow.cpp [hunk 1]
```diff
@@ -40,7 +40,10 @@ void genShadowForAccount(const struct spwd* spwd, QueryData& results) {
       r["password_status"] = "not_set";
     } else if (password[0] == '!' || password[0] == '*' || password[0] == 'x') {
       r["password_status"] = "locked";
-    } else {
+    } else if password == "") {
```
Comment (directionless) [inline osquery/tables/system/linux/shadow.cpp:43]:
```suggestion
    } else if (password == "") {
```

File: osquery/tables/system/linux/shadow.cpp [hunk 2]
```diff
@@ -40,6 +40,8 @@ void genShadowForAccount(const struct spwd* spwd, QueryData& results) {
       r["password_status"] = "not_set";
     } else if (password[0] == '!' || password[0] == '*' || password[0] == 'x') {
       r["password_status"] = "locked";
+    } else if (password == "") {
```
Comment (theopolis) [inline osquery/tables/system/linux/shadow.cpp:43]: nitpick, can you use `password.empty()`?

**Review by theopolis — CHANGES_REQUESTED** [review]

File: osquery/tables/system/linux/shadow.cpp (hunk 2 above)
Comment (directionless) [inline osquery/tables/system/linux/shadow.cpp:43]:
```suggestion
    } else if (password.empty()) {
```

Comment (directionless) [reply osquery/tables/system/linux/shadow.cpp]: Sure! The magic of github suggests means I don't even have to check out the code again =)

**Review by theopolis — APPROVED** [review]

**Pull request merged**